File: snippet-service/src/main/java/com/devflow/snippet/service/SnippetService.java
package com.devflow.snippet.service;

import java.util.List;

import com.devflow.snippet.dto.CreateSnippetRequest;
import com.devflow.snippet.dto.SnippetResponse;
import com.devflow.snippet.dto.UpdateSnippetRequest;

public interface SnippetService {

    SnippetResponse createSnippet(CreateSnippetRequest request, String userEmail);

    List<SnippetResponse> getAllSnippets(String userEmail);

    SnippetResponse getSnippetById(Long id, String userEmail);

    SnippetResponse updateSnippet(Long id,
                                  UpdateSnippetRequest request,
                                  String userEmail);

    void deleteSnippet(Long id, String userEmail);
}

File: devflow-frontend/src/api/snippetService.js
import axiosInstance from "./axiosInstance";

const getSnippets = () => {
  return axiosInstance.get("/api/snippets");
};

const createSnippet = (snippet) => {
  return axiosInstance.post("/api/snippets", snippet);
};

const updateSnippet = (id, snippet) => {
  return axiosInstance.put(`/api/snippets/${id}`, snippet);
};

const deleteSnippet = (id) => {
  return axiosInstance.delete(`/api/snippets/${id}`);
};

export default {
  getSnippets,
  createSnippet,
  updateSnippet,
  deleteSnippet,
};
